x-pack/plugin/esql/qa/testFixtures/src/main/resources/scoring.csv-spec

```diff
@@ -317,3 +317,43 @@ book_no:keyword | title:text
 2924            | A Gentle Creature and Other Stories: White Nights, A Gentle Creature, and The Dream of a Ridiculous Man (The World's Classics) | foobar        
 5948            | That We Are Gentle Creatures                                                                                                   | foobar
 ;
+
+
+semanticTextMatch
+required_capability: metadata_score
+required_capability: semantic_text_type
+required_capability: match_function
+
+from semantic_text metadata _id, _score
+| where match(semantic_text_field, "something")
+| sort _score desc
+| keep _id
+;
+
+_id:keyword
+2
+3
+1
+;
+
+semanticTextMatchWithAllTheTextFunctions
+
+required_capability: metadata_score
+required_capability: semantic_text_type
+required_capability: match_function
+required_capability: kql_function
+required_capability: qstr_function
+
+from semantic_text metadata _id, _score
+| where match(semantic_text_field, "something")
+        AND match(description, "some")
+        AND kql("description:some*")
+        AND NOT qstr("host:host1")
+| sort _score desc
+| keep _id
+;
+
+_id:keyword
```

Comment:
I dropped the score column for now because I am getting different values for single vs multi node tests 🤷‍♀️ .
It's a separate problem which we'll need to look into.

Example:
```

Actual: |  ----------------------------> these are the values generated with single node
-- | --
  | _id:keyword \| _score:double |  
  | 2           \| 5.603396578413904E18 |  
  | 3           \| 2.156063961865257E18 |  
  | 1           \| 8.411017936759685E17 |  
  |   |  
  | Expected: |     -------------------------------> these are the generated values for multi node
  | _id:keyword \| _score:double |  
  | 2           \| 5.603396E18 |  
  | 3           \| 2.156063E18 |  
  | 1           \| 8.411017E17

```